core-modules/000QubesVm.py

```diff
@@ -345,7 +346,14 @@ def __init__(self, **kwargs):
                 self.services['qubes-update-check'] = False
 
         # Initialize VM image storage class
-        self.storage = defaults["storage_class"](self)
+        self.storage = get_pool(self.pool_name, self).getStorage()
+        self.dir_path = self.storage.vmdir
+        self.icon_path = os.path.join(self.storage.vmdir, 'icon.png')
+        self.conf_file = os.path.join(self.storage.vmdir, self.name + '.conf')
+        self.appmenus_templates_dir = os.path.join(self.storage.vmdir, 'apps.templates')
+        self.appmenus_icons_dir = os.path.join(self.storage.vmdir, 'apps.icons')
```

Comment:
@marmarek
I tried that. I extended the [qubes-core-appmenus.py#L53-L65](https://github.com/QubesOS/qubes-core-admin-linux/blob/master/appmenus-scripts/qubes-core-appmenus.py#L53-L65) to access `self.storage.vmdir` inside the lambda. The `self.get_attrs_config()` in `QubesVm` is executed, before the `self.storage` is initialised. To initialise the `self.storage` you need to access the `pool_name` variable which is set by executing get_attrs_config().  

So we need to initialise the storage before we initialise appmenus.